MovingBlocks/Terasology
# feature(subsystems): extract DiscordRPCSubSystem

### Contains

New gradle-modules folder - `subsystems`

Extracted DiscordRPC subsystem - used by `PCFacade` only.

DiscordRPC was used in PlayerConfig directly - implemented config property observer. (like in rendering configs for runtime graphics settings changing)

### How to test

Required:
* Discord desktop

TestCases:
1. run game.
2. enable DIscord in settings
3. start game 
4. check status in Discord
5. rename your player name
6. check changes in discord (seems)

### Outstanding before merging

- [ ] improve subproject.settings.gradle and convert it in kotlin
- [ ] make configs composite( migrate to FlexableConfigs) - for removing discord config property from engine
- [x] (can be my dirty workspace) fix error with logs
